### ISSUE TITLE
Some parametric eq files have 0 as quality

### ISSUE THREAD

**Issue by PeterVerbeek** [jaakkopasanen/AutoEq#407]
Hi Jaakko,

As creator of the Peace equalizer interface for Equalizer APO I'm in the process of creating a simple and effective import feature to let a user easily select and use one of the many eq's on AutoEQ. For this I'm writing a piece of code to check and create a short version of the "result" folder on GitHub.

**Issue**
During this checking of the parametric eq filter files I found some that have the value of 0 for quality in one of the filter commands. This means an infinite bandwidth which is impossible. Besides, I'm not sure if Equalizer APO can cope with a 0 quality value (divide by zero issue or other issues).

**Remark**
I'm wondering about the usage and purpose of these 0 quality filter commands. Are they effective and how are they produced? Also, the 0 quality could be part of the fixed band eq files too. If needed I could check and confirm this.

**Possible solution**
For all purposes a quality value of 0.01 (almost infinite bandwidth) should suffice. 

**All 0 quality filter commands**
Filter 3: ON PK Fc 3980 Hz Gain -2.7 dB Q 0.00 Measurer crinacle Way ears-711_harman_over-ear_2018 Device Audeze LCD-24
Filter 3: ON PK Fc 3657 Hz Gain -3.9 dB Q 0.00 Measurer crinacle Way ears-711_harman_over-ear_2018 Device Audeze MX-4
Filter 2: ON PK Fc 1575 Hz Gain -1.1 dB Q 0.00 Measurer crinacle Way ears-711_harman_over-ear_2018 Device Audio-Technica ATH-M30
Filter 3: ON PK Fc 7254 Hz Gain -1.8 dB Q 0.00 Measurer crinacle Way ears-711_harman_over-ear_2018 Device Verum One
Filter 2: ON PK Fc 1855 Hz Gain -4.0 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Audeze LCD-MX4
Filter 10: ON PK Fc 18657 Hz Gain 0.9 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device BLON B20
Filter 3: ON PK Fc 5025 Hz Gain -3.0 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Denon AH-D7200
Filter 4: ON PK Fc 5564 Hz Gain -2.4 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Focal Utopia (sample 3)
Filter 4: ON PK Fc 7793 Hz Gain -2.3 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Nan-6 Nan-6
Filter 7: ON PK Fc 977 Hz Gain 0.5 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Pioneer Monitor 10
Filter 5: ON PK Fc 15549 Hz Gain -3.0 dB Q 0.00 Measurer crinacle Way gras_43ag-7_harman_over-ear_2018 Device Stax SR-X Mk2
Filter 2: ON PK Fc 2416 Hz Gain -1.6 dB Q 0.00 Measurer crinacle Way harman_in-ear_2019v2 Device AKG N5005 (reference)
Filter 5: ON PK Fc 4712 Hz Gain 2.2 dB Q 0.00 Measurer crinacle Way harman_in-ear_2019v2 Device EarSonics Velvet (warm)
Filter 5: ON PK Fc 16044 Hz Gain -3.1 dB Q 0.00 Measurer crinacle Way harman_in-ear_2019v2 Device Fearless Audio S10 Genie
Filter 1: ON PK Fc 15 Hz Gain -3.0 dB Q 0.00 Measurer crinacle Way harman_in-ear_2019v2 Device Nostalgia Audio Benbulbin
Filter 4: ON PK Fc 9088 Hz Gain 2.0 dB Q 0.00 Measurer crinacle Way harman_in-ear_2019v2 Device Tin HiFi T5
Filter 1: ON PK Fc 18 Hz Gain 4.5 dB Q 0.00 Measurer headphonecom Way headphonecom_harman_over-ear_2018 Device Grado R225
Filter 5: ON PK Fc 14362 Hz Gain 7.1 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_in-ear_2019v2 Device Polk Audio UltraFit 1000
Filter 5: ON PK Fc 8089 Hz Gain 1.4 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_over-ear_2018 Device AKG K267 Tiesto (stage)
Filter 4: ON PK Fc 4414 Hz Gain -3.8 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_over-ear_2018 Device Audio-Technica ATH-W5000 2013
Filter 5: ON PK Fc 15040 Hz Gain -2.0 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_over-ear_2018 Device Dan Clark Audio Aeon Flow Closed
Filter 5: ON PK Fc 16653 Hz Gain -1.4 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_over-ear_2018 Device Sennheiser HD 580
Filter 4: ON PK Fc 3432 Hz Gain -3.8 dB Q 0.00 Measurer innerfidelity Way innerfidelity_harman_over-ear_2018 Device Stax SR-Sigma 07381
Filter 5: ON PK Fc 17359 Hz Gain -3.1 dB Q 0.00 Measurer oratory1990 Way harman_in-ear_2019v2 Device Sennheiser IE 800 S
Filter 4: ON PK Fc 5387 Hz Gain -3.7 dB Q 0.00 Measurer oratory1990 Way harman_over-ear_2018 Device Gold Planar GL2000
Filter 3: ON PK Fc 3053 Hz Gain -1.9 dB Q 0.00 Measurer oratory1990 Way harman_over-ear_2018 Device Verum One
Filter 3: ON PK Fc 2920 Hz Gain -1.1 dB Q 0.00 Measurer referenceaudioanalyzer Way referenceaudioanalyzer_hdm-x_harman_over-ear_2018 Device Sennheiser HD 660 S
Filter 9: ON PK Fc 15188 Hz Gain -1.1 dB Q 0.00 Measurer referenceaudioanalyzer Way referenceaudioanalyzer_hdm1_harman_over-ear_2018 Device V-MODA Crossfade LP
Filter 1: ON PK Fc 36 Hz Gain -1.2 dB Q 0.00 Measurer referenceaudioanalyzer Way referenceaudioanalyzer_siec_harman_in-ear_2019v2 Device Lime Ears Model X (on)
Filter 5: ON PK Fc 18851 Hz Gain 2.3 dB Q 0.00 Measurer rtings Way rtings_harman_in-ear_2019v2 Device Anker SoundCore Life 2

**Comment by jaakkopasanen** [comment]
Hello Peter!

I'm very delighted to hear you working on this.

The parametric equalizer settings are produced automatically with a gradient descent optimization algorithm. This algorithm currently doesn't have constraints for Q values and such is some rare cases might produce filters which have Q of 0.0 if the optimizer sees this as the easiest way to achieve minimum error.

The 0 Q filters should have inifinite bandwidth meaning they are basically just preamp. You could add the gains from these filters to preamp and then omit the filters. [BLON B20](https://github.com/jaakkopasanen/AutoEq/tree/master/results/crinacle/gras_43ag-7_harman_over-ear_2018/BLON%20B20) for example has 0 Q filter with 0.9 dB of gain and the preset asks for -6.9 dB of preamp. You should be able to just change the preamp to -6 dB and omit the 18.7 kHz filter entirely.

I do have plans to fix these things in the parametric eq optimizer but this turned out to be more difficult than I anticipated.

**Comment by PeterVerbeek** [comment]
Yes, I came up with the same idea: adding to the pre amp. For me it's very easy to do in the (small) checking and compressing tool I'm building. So I'm wondering if you would like me to correct these files the tool finds (including the fixed band ones) and send them to you? I guess manually overwriting some files in the result folder could mess up the automatic creation of this folder (if it's implemented that way).

**Comment by jaakkopasanen** [comment]
Thanks for the offer but I'd rather keep everything automated. With the number of measurements in AutoEq, it becomes imposible to manage these things manually. So for the time being we'll have to just accept that there are some results which have some problems.

**Comment by PeterVerbeek** [comment]
I completely understand. Thanks for replying on this matter. I'll correct it at my side of things. Anyway, it is a pleasure to talk to you.

**Comment by bhack** [comment]
Probably we could document this somewhere especially if we could remove manually a little bit of global preamp attenuation.

**Comment by jaakkopasanen** [comment]
There is an older issue about this very same problem #253 so I'll close this one as duplicate. Follow the other one if you wish to keep updated.
